client-spark/common/src/main/java/org/apache/spark/shuffle/RssSparkConfig.java

```diff
@@ -278,4 +281,17 @@ public Double apply(String in) {
   public static TypedConfigBuilder<String> createStringBuilder(ConfigBuilder builder) {
     return builder.stringConf();
   }
+
+  public static RssConf toRssConf(SparkConf sparkConf) {
```

Comment:
Why change conf design in this ZSTD PR?

Reply:
I want to make compressorFactory accessed by MR and Spark to create concrete codec which will be initialized by specified conf, so it will have two choice.
1. Use the shareable RssConf like this PR
2. Introduce the extra config bean of compression (I think there is no need to do so)

Besides, I want to refactor the code of MR/Spark client conf entry, this PR is to do some partial work. Please refer to #200